src/config/server.rs

```diff
@@ -94,6 +94,11 @@ pub struct Server {
     /// `workers` sets the number of worker threads. @default the number of
     /// system cores.
     pub workers: Option<usize>,
+
+    #[serde(default, skip_serializing_if = "is_default")]
+    /// When set to `true`, it will ensure no HTTP, GRPC, or any other IO call
+    /// is made more than once within the context of a single GraphQL request.
+    pub dedupe_io: Option<bool>,
```

Comment:
A bit of change is required here:
- [ ] Rename from `dedupe_io` to `dedupe`
- [ ] Move the setting from `server` to `upstream`